File: examples/javascript/pricing.js
/**
 * VeeoCore API - Exemple JavaScript
 * Calcul de prix pour un trajet VTC
 */

const API_BASE = 'https://api-core.veeo-stras.fr/api/v1';
const API_KEY = 'YOUR_API_KEY'; // Remplacez par votre clé API

async function calculatePrice(origin, destination) {
  try {
    const response = await fetch(`${API_BASE}/pricing/calculate`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'X-API-Key': API_KEY
      },
      body: JSON.stringify({ origin, destination })
    });

    if (!response.ok) {
      throw new Error(`HTTP ${response.status}: ${response.statusText}`);
    }

    const data = await response.json();
    return data;
  } catch (error) {
    console.error('Erreur API:', error.message);
    throw error;
  }
}

// Exemple d'utilisation
async function main() {
  const origin = {
    lat: 48.5734,
    lng: 7.7521,
    address: 'Gare de Strasbourg'
  };

  const destination = {
    lat: 48.5853,
    lng: 7.7350,
    address: 'Aéroport de Strasbourg-Entzheim'
  };

  console.log('Calcul du prix...');
  const result = await calculatePrice(origin, destination);

  console.log('\n📍 Trajet:');
  console.log(`   ${result.distance_km} km - ${result.duration_min} min`);

  console.log('\n💰 Prix par véhicule:');
  result.prices.forEach(vehicle => {
    console.log(`   ${vehicle.name}: ${vehicle.price}€`);
  });
}

main();

File: examples/python/pricing.py
"""
VeeoCore API - Exemple Python
Calcul de prix pour un trajet VTC
"""

import requests
import json

API_BASE = 'https://api-core.veeo-stras.fr/api/v1'
API_KEY = 'YOUR_API_KEY'  # Remplacez par votre clé API


def calculate_price(origin: dict, destination: dict) -> dict:
    """Calcule le prix d'un trajet VTC."""
    response = requests.post(
        f'{API_BASE}/pricing/calculate',
        headers={
            'Content-Type': 'application/json',
            'X-API-Key': API_KEY
        },
        json={
            'origin': origin,
            'destination': destination
        }
    )
    response.raise_for_status()
    return response.json()


def main():
    origin = {
        'lat': 48.5734,
        'lng': 7.7521,
        'address': 'Gare de Strasbourg'
    }

    destination = {
        'lat': 48.5853,
        'lng': 7.7350,
        'address': 'Aéroport de Strasbourg-Entzheim'
    }

    print('Calcul du prix...')
    result = calculate_price(origin, destination)

    print(f"\n📍 Trajet: {result['distance_km']} km - {result['duration_min']} min")

    print('\n💰 Prix par véhicule:')
    for vehicle in result['prices']:
        print(f"   {vehicle['name']}: {vehicle['price']}€")


if __name__ == '__main__':
    main()
